### ISSUE TITLE
Fail if the function `plot_contour()` specifies a pair of parameters that cannot coexist

### ISSUE THREAD

**Issue by kAIto47802** [optuna/optuna#5624]
### Expected behavior

Currently, if a pair of parameters, which cannot co-exist in `trial.params`, is specified, it will fail with an unfriendly error message.
Note that when `params=None` is specified, as all pairs are used in the plot, anyways the function fails.
There are two possible solutions to this:

- Raise an error saying that the plot cannot be performed, or
- Return an empty plot without an error.

Please see the additional context for more information.

### Environment

- Optuna version: master branch
- Python version: any
- OS: any

### Error messages, stack traces, or logs

```python
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[8], line 5
      2 study = optuna.create_study(sampler=sampler)
      3 study.optimize(objective_single_dynamic_with_categorical, n_trials=30)
----> 5 fig = optuna.visualization.plot_contour(study, params=["x1", "x2"])
      6 fig.show()

File ~/Documents/2_PFN/pfn-workspace/optuna/optuna/visualization/_contour.py:122, in plot_contour(study, params, target, target_name)
    120 _imports.check()
    121 info = _get_contour_info(study, params, target, target_name)
--> 122 return _get_contour_plot(info)

File ~/Documents/2_PFN/pfn-workspace/optuna/optuna/visualization/_contour.py:140, in _get_contour_plot(info)
    138 y_param = sorted_params[1]
    139 sub_plot_info = sub_plot_infos[0][0]
--> 140 sub_plots = _get_contour_subplot(sub_plot_info, reverse_scale, target_name)
    141 figure = go.Figure(data=sub_plots, layout=layout)
    142 figure.update_xaxes(title_text=x_param, range=sub_plot_info.xaxis.range)

File ~/Documents/2_PFN/pfn-workspace/optuna/optuna/visualization/_contour.py:228, in _get_contour_subplot(info, reverse_scale, target_name)
    225 xys = np.array(list(info.z_values.keys()))
    226 zs = np.array(list(info.z_values.values()))
--> 228 z_values[xys[:, 1], xys[:, 0]] = zs
    230 if len(x_indices) < 2 or len(y_indices) < 2:
    231     return go.Contour(), go.Scatter(), go.Scatter()

IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```

### Steps to reproduce

The following code yields the error above:

```python
import optuna

def objective(trial: optuna.Trial) -> float:
    category = trial.suggest_categorical("category", ["foo", "bar"])
    if category == "foo":
        return (trial.suggest_float("x1", 0, 10) - 2) ** 2
    else:
        return -((trial.suggest_float("x2", -10, 0) + 5) ** 2)

study = optuna.create_study()
study.optimize(objective, n_trials=30)

fig = optuna.visualization.plot_contour(study, params=["x1", "x2"])
fig.show()
```

> [!NOTE]
> In the code above, since `x1` and `x2` cannot be defined at the same time, the function values cannot be defined in the space composed of `x1` and `x2` and contour cannot be plot.

### Additional context (optional)

This issue is related to the PR:

- #5599 

This PR simlpy suppressed the issue reported here and left TODO comments saying that this issue should be addressed in the future.

https://github.com/optuna/optuna/blob/1925ca1a051b3bf11819478b65d830de614774b2/tests/visualization_tests/test_visualizations.py#L90-L92

In these lines, if the specified parameter pair cannot exist at the same time, `z_values` becomes empty.

https://github.com/optuna/optuna/blob/1925ca1a051b3bf11819478b65d830de614774b2/optuna/visualization/_contour.py#L341-L344

An empty `info.z_values` causes an error.

https://github.com/optuna/optuna/blob/1925ca1a051b3bf11819478b65d830de614774b2/optuna/visualization/_contour.py#L228

As mentioned earlier, one solution is to return an empty plot without an error, meaning that the current code 

https://github.com/optuna/optuna/blob/1925ca1a051b3bf11819478b65d830de614774b2/optuna/visualization/_contour.py#L203-L231

should be fixed like:

```python
def _get_contour_subplot(
    info: _SubContourInfo,
    reverse_scale: bool,
    target_name: str = "Objective Value",
) -> tuple["Contour", "Scatter", "Scatter"]:
    x_indices = info.xaxis.indices
    y_indices = info.yaxis.indices

    if len(x_indices) < 2 or len(y_indices) < 2 or len(info.z_values) == 0:
        return go.Contour(), go.Scatter(), go.Scatter()
```

> [!NOTE]
> As long as the early return is defined before this line, any change is acceptable.
> ```python
> z_values[xys[:, 1], xys[:, 0]] = zs
> ```

**Comment by nabenabe0928** [comment]
Thank you for raising the issue!
We can simply address this issue by Solution 2, i.e., to return an empty plot, with warning:)

**Comment by not522** [comment]
This issue has been resolved by #5630.
